pkg/network/ebpf/c/protocols/kafka/kafka-parsing.h

```diff
@@ -1828,26 +1835,34 @@ static __always_inline void update_topic_name_size_telemetry(kafka_telemetry_t *
     __sync_fetch_and_add(&kafka_tel->topic_name_size_buckets[bucket_idx], 1);
 }
 
-// update_classified_fetch_api_version_hits_telemetry updates the array keeping track of fetch api versions hits in the telemetry
-static __always_inline void update_classified_fetch_api_version_hits_telemetry(kafka_telemetry_t *kafka_tel, __u8 api_version) {
-    __u8 bucket_idx = api_version; // Note that api_version may be 0
-
-    if (bucket_idx < 0 || bucket_idx > KAFKA_CLASSIFICATION_MAX_SUPPORTED_FETCH_REQUEST_API_VERSION) {
+// update_classified_produce_api_version_hits_telemetry updates the array keeping track of produce api versions hits in the telemetry
+static __always_inline void update_classified_produce_api_version_hits_telemetry(kafka_telemetry_t *kafka_tel, __s16 version) {
+    if (version < 0 || version > KAFKA_CLASSIFICATION_MAX_SUPPORTED_PRODUCE_REQUEST_API_VERSION) {
         return;
     }
 
-    __sync_fetch_and_add(&kafka_tel->classified_fetch_api_version_hits[bucket_idx], 1);
-}
+    // Ensure that the bucket index falls within the valid range.
+    __u8 bucket_idx = version % (KAFKA_CLASSIFICATION_MAX_SUPPORTED_PRODUCE_REQUEST_API_VERSION + 1);
```

Comment:
this line is weird for the average reader, perhaps mention this is to please the verifier